### ISSUE TITLE
Implement custom MarshalJSON and  UnmarshalJSON for xdr.ClaimPredicate

### ISSUE THREAD

**Issue by abuiles** [stellar/go#2932]
### What problem does your feature solve?

We are storing `xdr.ClaimPredicate` in the DB as the `base64` encoded value of the XDR which is fine and it helps us to verify state, we'll need a different representation for `xdr.ClaimPredicate` when rendering via the web api.

### What would you like to see?

Given a `xdr.ClaimPredicate` we should be able to Marshal it to JSON using the following representation

```
 {
  "and" : [
    {"after": 1594066025},
    {"before": 1594066090}
  ]
}
```

`ClaimPredicateTypeClaimPredicateUnconditional` should be `{}`.

At the moment, this can be worked on top of this branch https://github.com/abuiles/go/tree/get-claimable-balance-by-id which contains the resource adapter for claimable balances

**Comment by abuiles** [comment]
@tamirms we also need to update the predicate value in the operation details processor and effects processor.

**Comment by bartekn** [comment]
Fixed in #2947.
